source/adapters/level_zero/command_buffer.cpp

```diff
@@ -764,6 +772,41 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferEnqueueExp(
     ZE2UR_CALL(zeCommandListAppendBarrier,
                (SignalCommandList->first, RetEvent->ZeEvent, 1,
                 &(CommandBuffer->SignalEvent->ZeEvent)));
+
+    if ((Queue->Properties & UR_QUEUE_FLAG_PROFILING_ENABLE)) {
+      // Multiple submissions of a command buffer implies that we need to save
+      // the event timestamps before resubmiting the command buffer. We
+      // therefore copy the these timestamps in a dedicated USM memory section
+      // before completing the command buffer execution, and then attach this
+      // memory to the event returned to users to allow to allow the profiling
+      // engine to recover these timestamps.
+      ur_usm_desc_t USMDesc{};
+      ur_usm_device_desc_t UsmDeviceDesc{};
+      UsmDeviceDesc.stype = UR_STRUCTURE_TYPE_USM_DEVICE_DESC;
+      ur_usm_host_desc_t UsmHostDesc{};
+      UsmHostDesc.stype = UR_STRUCTURE_TYPE_USM_HOST_DESC;
+      UsmDeviceDesc.pNext = &UsmHostDesc;
+      USMDesc.pNext = &UsmDeviceDesc;
+      USMDesc.align = 4; // 4byte-aligned
+
+      size_t Size = WaitEventList.size() * sizeof(ze_kernel_timestamp_result_t);
+
+      struct command_buffer_profiling_t *Profiling =
+          new command_buffer_profiling_t();
+
+      Profiling->NumEvents = WaitEventList.size();
+
+      urUSMSharedAlloc(RetEvent->Context, CommandBuffer->Device, &USMDesc,
+                       nullptr, Size, (void **)&Profiling->Timestamps);
```

Comment:
The implementation has been updated to make it use host only memory instead of USMShared since `zeCommandListAppendQueryKernelTimestamps` seems to be able to copy directly to host memory.

Reply:
Can we guarantee that in all cases if this is the usage wording?

> The application must ensure the memory pointed to by both dstptr and pOffsets is accessible by the device on which the command list was created.

Reply:
Trying to read up on L0 USM memory support https://spec.oneapi.io/level-zero/latest/core/PROG.html#types 
and turns out the minimum capabilities are actually different from SYCL/USM. I incorrectly assumed they were the same, but L0 devices do always support shared USM. So I think we can use your previous approve using the shared USM allocation

## L0 support matrix
![image](https://github.com/oneapi-src/unified-runtime/assets/2334151/cad1a296-b20e-4e18-969e-f87b247db46d)

## SYCL/OpenCL support matrix
![image](https://github.com/oneapi-src/unified-runtime/assets/2334151/b64ab9d2-5ac7-4a8b-ab72-271ed258fd33)